java/rocksjni/compaction_job_info.cc

```diff
@@ -28,8 +27,8 @@ jlong Java_org_rocksdb_CompactionJobInfo_newCompactionJobInfo(
  * Method:    disposeInternal
  * Signature: (J)V
  */
-void Java_org_rocksdb_CompactionJobInfo_disposeInternal(
-    JNIEnv*, jobject, jlong jhandle) {
+void Java_org_rocksdb_CompactionJobInfo_disposeInternal(JNIEnv*, jobject,
```

Comment:
@sagar0 I really hate this style of indenting on the JNI signature declarations. I am not sure it makes sense as the method names will always be very very long. Ultimately this leads to the downside of having one variable a line, difficult to read code and no benefit.

Is there any chance to get an exclusion for this rule for CPP code inside `java/rocksjni`?

Reply:
@adamretter We are just using the clang formatter. `make format` uses https://github.com/facebook/rocksdb/blame/master/build_tools/format-diff.sh which internally calls clang-format. (Our internal linter is much more stricter than the formatter, whining on other rules.)
While we could potentially have an exclusion for the files under java directory, we will lose the efficiency of formatting everything with a tool, leading to many different formatting styles from different contributors based on personal preferences. I am open to suggestions.

Reply:
> I am not sure it makes sense as the method names will always be very very long. Ultimately this leads to the downside of having one variable per line, difficult to read code and no benefit.

Clang format does not always put one arg per line (if this is your main concern). You can see that when there are many params, clang chooses a different style of putting multiple args on the same line. Example:
https://github.com/facebook/rocksdb/blob/bb474e9a02abfe9c97b193a27c846de9fbdb4389/db/builder.cc#L65-L82

Reply:
My point was that it does put each variable on a separate line when the method name is very very long which is typically only the case in the JNI C++ code. I don't want to turn off formatting for the JNI C++ code, I just want to tweak it for that folder so that we have sensible formatting that is both readable and usable. Is it possible to configure the clang format rules for RocksDB or add exceptions for only certain rules?

Reply:
I am still not convinced that we should tweak the formatting for the JNI code. There are many places in the current C++ API code where each variable is placed on a separate line, as the method name sometimes becomes long when prefixed with the class name; granted it could be a little more so in the JNI C++ code. This is coming due to the application of [this style rule](https://google.github.io/styleguide/cppguide.html#Function_Declarations_and_Definitions) -- which seems sane enough to me.

But in case if it is decided to tweak, [AlignAfterOpenBracket](http://clang.llvm.org/docs/ClangFormatStyleOptions.html) could be set to `DontAlign` or `AlwaysBreak` for JNI files, I believe (I am not an expert on style-rules, just googled my way to it).